ibis/backends/polars/compiler.py

```diff
@@ -1203,9 +1203,19 @@ def execute_union(op, **kw):
     return result
 
 
+def bit_cast_uint64_to_int64(uint: pl.Series) -> pl.Series:
+    happy_path = uint.cast(pl.Int64, strict=False)
+    in_range = uint - (1 << 63)
+    casted = in_range.cast(pl.Int64, strict=False)
+    corrected = casted - ((1 << 63) - 1)
+    INT64_MAX = 2**63 - 1
+    return pl.when(uint <= INT64_MAX).then(happy_path).otherwise(corrected)
```

Comment:
This doesn't handle wraparound conversion the way you probably intended (yours changes the total ordering at the split point) - ideally converting a `uint64` to an `int64` is just a cast and an addition of `INT64_MIN`. 

Try this instead:

```suggestion
    sub_then_cast = (uint - 2 ** 63).cast(pl.Int64, strict=False)
    cast_then_sub = uint.cast(pl.Int64, strict=False) + (-2 ** 63)
    return sub_then_cast.fill_null(cast_then_sub)
```

Comparison:

```
┌──────────────────────┬──────────────────────┬──────────────────────┐
│ x                    ┆ yours                ┆ mine                 │
│ ---                  ┆ ---                  ┆ ---                  │
│ u64                  ┆ i64                  ┆ i64                  │
╞══════════════════════╪══════════════════════╪══════════════════════╡
│ 18446744073709551615 ┆ 0                    ┆ 9223372036854775807  │
│ 18446744073709551614 ┆ -1                   ┆ 9223372036854775806  │
│ 18446744073709551613 ┆ -2                   ┆ 9223372036854775805  │
│ 9223372036854775809  ┆ -9223372036854775806 ┆ 1                    │
│ 9223372036854775808  ┆ -9223372036854775807 ┆ 0                    │
│ 9223372036854775807  ┆ 9223372036854775807  ┆ -1                   │
│ 2                    ┆ 2                    ┆ -9223372036854775806 │
│ 1                    ┆ 1                    ┆ -9223372036854775807 │
│ 0                    ┆ 0                    ┆ -9223372036854775808 │
└──────────────────────┴──────────────────────┴──────────────────────┘
```

Reply:
Since this is taking a hash as input, the ordering of the input isn't meaningful, so I wasn't trying to maintain ordering. Only trying to maintain the entropy. I should probably rename the function. What do you think, should we care about order? I think we should still consider performance though, so we should break ties using that

Reply:
I don't think we should care about order, just noting that the implementation I posted here does things the same way a reinterpret cast would, while yours does something different. As long as it's a 1:1 mapping between uint64 -> int64 then I don't think we care what it is as long as it's efficient enough.

Reply:
Actually, turns out polars has a builtin for this too - can just do `col.reinterpret(signed=True)`. Much cleaner.

Reply:
excellent, thank you!